Loop/Managers/DeviceDataManager.swift

```diff
@@ -1400,6 +1400,68 @@ extension DeviceDataManager: SupportInfoProvider {
     
 }
 
+//MARK: TherapySettingsViewModelDelegate
+
+extension DeviceDataManager: TherapySettingsViewModelDelegate {
+    
+    func syncBasalRateSchedule(items: [RepeatingScheduleValue<Double>], completion: @escaping (Swift.Result<BasalRateSchedule, Error>) -> Void) {
+        pumpManager?.syncBasalRateSchedule(items: items, completion: completion)
+    }
+    
+    func syncDeliveryLimits(deliveryLimits: DeliveryLimits, completion: @escaping (Swift.Result<DeliveryLimits, Error>) -> Void) {
+        // FIRST we need to check to make sure if we have to cancel temp basal first
+        loopManager.maxTempBasalSavePreflight(unitsPerHour: deliveryLimits.maximumBasalRate?.doubleValue(for: .internationalUnitsPerHour)) { [weak self] error in
+            if let error = error {
+                completion(.failure(SaveTherapySettingsError.cancelTempBasalError(error)))
+            } else if let pumpManager = self?.pumpManager {
+                pumpManager.syncDeliveryLimits(limits: deliveryLimits, completion: completion)
+            } else {
+                completion(.failure(SaveTherapySettingsError.missingPumpManager))
+            }
+        }
+    }
+    
+    func saveCompletion(for therapySetting: TherapySetting, therapySettings: TherapySettings) {
+        switch therapySetting {
+        case .glucoseTargetRange:
+            loopManager.mutateSettings { settings in settings.glucoseTargetRangeSchedule = therapySettings.glucoseTargetRangeSchedule }
+        case .preMealCorrectionRangeOverride:
+            loopManager.mutateSettings { settings in settings.preMealTargetRange = therapySettings.correctionRangeOverrides?.preMeal }
+        case .workoutCorrectionRangeOverride:
+            loopManager.mutateSettings { settings in settings.legacyWorkoutTargetRange = therapySettings.correctionRangeOverrides?.workout }
+        case .suspendThreshold:
+            loopManager.mutateSettings { settings in settings.suspendThreshold = therapySettings.suspendThreshold }
+        case .basalRate:
+            loopManager.basalRateSchedule = therapySettings.basalRateSchedule
+        case .deliveryLimits:
```

Comment:
I'd like to keep this here, since it was just moved as a whole from `StatusTableViewController` (where I'd argue it didn't belong in the first place).  This method of saving "completion" has been around for a long while, let's keep it the way it is for now?